rsocket/RSocketClient.cpp

```diff
@@ -116,13 +117,24 @@ folly::Future<folly::Unit> RSocketClient::resume() {
   });
 }
 
-void RSocketClient::disconnect(folly::exception_wrapper ex) {
-  CHECK(stateMachine_);
-  evb_->runInEventBaseThread(
-      [ stateMachine = stateMachine_, ex = std::move(ex) ]() mutable {
-        VLOG(2) << "Disconnecting RSocketStateMachine on EventBase";
-        stateMachine->disconnect(std::move(ex));
-      });
+folly::Future<folly::Unit> RSocketClient::disconnect(
+    folly::exception_wrapper ew) {
+  if (!stateMachine_) {
```

Comment:
This should probably be a CHECK().  If I remember correctly, you cannot create a RSocketClient instance without creating a stateMachine_

Reply:
What about returning an Error future in this case instead?

Reply:
Oh whoops, this is from an older part of my diff when I was (mistakenly) moving the state machine into the disconnect lambda.  Let me fix that.

Reply:
@dymk I made it a `DCHECK`, as this is kind of an invariant violation on our part.  I guess I could make it return an exception just in case, think it's worth it?

Reply:
Well, it's not an invariant that we can enforce internally, as the application is the one with a direct handle on the client. So it's probably worth it to throw something catchable rather than something that aborts the program.

Reply:
Works for me.